modules/dynamic/test/src-jvm/smithy4s/dynamic/EnumSpec.scala

```diff
@@ -223,24 +228,29 @@ class EnumSpec extends FunSuite {
   test("Smithy 2.0 enum members get their hints compiled") {
     assertEnum(
       ShapeId("example", "EnumWithTraits"),
-      expectedValues = List(
-        EnumValue(
-          stringValue = "FIRE",
-          intValue = 0,
-          value = 0,
-          name = "FIRE",
-          hints = Hints.empty
-        ),
-        EnumValue(
-          stringValue = "ICE",
-          intValue = 1,
-          value = 1,
-          name = "ICE",
-          hints = Hints(
-            ShapeId("smithy.api", "deprecated") -> Document.obj()
+      expectedValues = index =>
+        List(
+          EnumValue(
+            stringValue = "FIRE",
+            intValue = 0,
+            value = 0,
+            name = "FIRE",
+            hints = Hints.empty
+          ),
+          EnumValue(
+            stringValue = "ICE",
+            intValue = 1,
+            value = 1,
+            name = "ICE",
+            hints = Hints(
+              (
+                ShapeId("smithy.api", "deprecated"),
+                Document.obj(),
+                index.getSchema(api.Deprecated.id).get
```

Comment:
I don't really like that the addition of a new field suddenly suddenly imposes an arbitrary need for some external reference, when so far things could be built in a somewhat inductive way. 

In particular, I feel like this added constraint closes the door the the opportunity to build data-description  some other way than building a bonafide smithy model. For instance, the current model would allow for parsing schemas from less-constrained IDLs, such as json-schema, and capture hints as json-schema freeform extensions that wouldn't be tied to smithy4s schemas

In other words, I believe the concept of smithy4s schema needs to remain somewhat decoupled from the concept of smithy model and the constraints that apply to smithy models should not necessarily be transposed to smithy4s schemas. 

In all frankness, I think it's a rather annoying mistake from me to have left the data-structure of `Hints.Binding` publicly inspectable, and I'd be in favour of closing it for 0.18 

I do however agree that there are some interesting usecases that would be allowed by a ref held by `Hints.Binding` that would allow to retrieve a schema when the circumstances allow for it : `def getSchema : Option[Schema[_]]`. This would ease my worry about being able to accommodate other usecases than bonafide smithy-schemas, whilst offering more flexibility to traverse schemas as if there was an actual model backing them up.

Reply:
> it makes sense, but seems a bit like a "what if" scenario rather than something we're about to start doing.

I actually may have a usecase internally 😅 

Also, it's no more "what if" than any hypothetical tool that'd leverage the API you're adding in this PR (once again : the display of protocols supported by a service is not motivating enough to enforce the constraint, in particular as it'd be solve by the presence of a `def : Option[Schema[_]]` on `Hints.Binding`